pkg/cluster/k8sres.go

```diff
@@ -150,42 +150,53 @@ func makeLogicalBackupResources(config *config.Config) acidv1.Resources {
 func (c *Cluster) enforceMinResourceLimits(resources *v1.ResourceRequirements) error {
 	var (
 		isSmaller bool
-		err       error
 		msg       string
 	)
 
 	// setting limits too low can cause unnecessary evictions / OOM kills
 	cpuLimit := resources.Limits[v1.ResourceCPU]
 	minCPULimit := c.OpConfig.MinCPULimit
 	if minCPULimit != "" {
-		isSmaller, err = util.IsSmallerQuantity(cpuLimit.String(), minCPULimit)
+		minCPU, err := resource.ParseQuantity(minCPULimit)
 		if err != nil {
-			return fmt.Errorf("could not compare defined CPU limit %s for %q container with configured minimum value %s: %v",
-				cpuLimit.String(), constants.PostgresContainerName, minCPULimit, err)
+			return fmt.Errorf("could not parse min CPU limit %s: %v", minCPULimit, err)
 		}
-		if isSmaller {
-			msg = fmt.Sprintf("defined CPU limit %s for %q container is below required minimum %s and will be increased",
-				cpuLimit.String(), constants.PostgresContainerName, minCPULimit)
-			c.logger.Warningf(msg)
-			c.eventRecorder.Eventf(c.GetReference(), v1.EventTypeWarning, "ResourceLimits", msg)
-			resources.Limits[v1.ResourceCPU], _ = resource.ParseQuantity(minCPULimit)
+		if !minCPU.IsZero() {
```

Comment:
this extra code is not required. If c.OpConfig.MinCPULimit is not defined it will be "". Even when it's "0" and the comparison code after isSmaller is executed if won't get enforced even when no limits are specified.